lib/core/src/animation/animator.rs

```diff
@@ -0,0 +1,51 @@
+use crate::system::web::AnimationFrameLoop;
+
+use nalgebra::zero;
+
+// ====================
+// === AnimatorData ===
+// ====================
+
+struct AnimatorData {
+    closure          : Box<dyn FnMut(f32)>,
+    previous_time    : Option<f32>,
+    step_duration    : f32,
+    accumulated_time : f32
+}
+
+impl AnimatorData {
+    pub fn new<F:FnMut(f32) + 'static>(steps_per_second:f32, f:F) -> Self {
```

Comment:
Let's not keep `'static` naked in types. There is a risk it populates to different functions and poisons a lot of code. Sometimes after refactoring it would not be needed anymore, but no one would notice that. Please create a type alias for such complex types always.

Reply:
Created this:
```rust
pub trait FnAnimation = FnMut(f32) + 'static;
```